### DISCUSSION TITLE
sunshine on Wayland from the flatpak

### DISCUSSION THREAD

**Discussion by paju1986** [LizardByte/community#106] [Sunshine] [answered]
im trying to run sunshine from flatpak on wayland but i dont manage to get it working i get this output:

phone:~$ flatpak run dev.lizardbyte.sunshine
Cannot load libcuda.so.1
[2022:12:16:19:49:36]: Error: Couldn't load cuda: -1
[2022:12:16:19:49:36]: Error: Failed to gain CAP_SYS_ADMIN
[2022:12:16:19:49:36]: Error: Failed to gain CAP_SYS_ADMIN
[2022:12:16:19:49:36]: Error: Couldn't get handle for DRM Framebuffer [86]: Possibly not permitted: do [sudo setcap cap_sys_admin+p sunshine]
[2022:12:16:19:49:36]: Error: Environment variable WAYLAND_DISPLAY has not been defined

reading the docs i find out this command:

sudo setcap cap_sys_admin+p $(readlink -f $(which sunshine))

but it doesnt seem to work, as when running "which sunshine" separately it doesnt found the sunshine binary. I guess it has to do with the flapak being sandboxing or something.
Can someone help here?

**Comment by brad-richardson** [comment, marked as answer]
Did you follow all the steps here? https://docs.lizardbyte.dev/projects/sunshine/en/latest/about/installation.html#flatpak-package

Make sure you run with `sudo -i PULSE_SERVER=unix:$(pactl info | awk '/Server String/{print$3}') flatpak run dev.lizardbyte.sunshine `

**Reply by kevindurb** [reply]
How do you do this as a systemd service?

**Reply by JoeneckSpoeneck** [reply]
Hey, when I run that command, I get this output:
`error: XDG_RUNTIME_DIR is invalid or not set in the environment.`
`bwrap: Can't find source path /run/user/1000/pulse/native: Permission denied`
I looked into the "pulse" directory:
`ll /run/user/1000/pulse             
insgesamt 4,0K`
`srw-rw-rw- 1 joeneckspoeneck joeneckspoeneck 0 31. Jan 21:01 native`
`-rw-r--r-- 1 joeneckspoeneck joeneckspoeneck 5 31. Jan 21:01 pid`
Do you have any idea what's wrong?

**Reply by XT-Martinez** [reply]
I'm also encountering this error after upgrading my fedora 39 install. Permission denied on /run/user/1000/pulse/native

**Reply by JumperLuko** [reply]
Same problem on Fedora 39

**Reply by XT-Martinez** [reply]
It turns out to be an issue with the bwrap that ships with fedora. As a workaround, you can set `chmod u+s /usr/bin/bwrap`.

**Reply by MinimaxHeadroom** [reply]
> It turns out to be an issue with the bwrap that ships with fedora. As a workaround, you can set `chmod u+s /usr/bin/bwrap`.

I'm already cautious about trying such hackish workarounds. Nevertheless, the command won't work on atomic desktop OSes like Fedora Silverblue: `chmod: changing permissions of '/usr/bin/bwrap': Read-only file system`.

I've tried both system and user-level installs of the Sunshine Flatpak on Silverblue 39 with Wayland (GPU: Nvidia Quadro T1000), following the steps in the official documentation, but Sunshine (0.21.0.5bca024.dirty) continues to report `Fatal: Couldn't find any working encoder` in its logs.

Stuck at this point.

**Reply by XT-Martinez** [reply]
For Silverblue, you can copy `/usr/bin/bwrap` to another writable directory (e.g. `/usr/local/bin`) and then override flatpak's bwrap binary using `FLATPAK_BWRAP=/path/to/bwrap flatpak run dev.lizardbyte.sunshine`

**Reply by XT-Martinez** [reply]
I'm also on Silverblue and I did not encounter this issue after a fresh install but after upgrading the system, I got this permission denied issue. Either something changed with the bubblewrap package or the newer kernel got more restrictive.

**Reply by linuxkernel94** [reply]
I'm facing the same issue on Fedora silverblue 40, I couldn't find a solution yet

**Reply by JumperLuko** [reply]
I tried everything, but I needed to use RPM version, because changing bwrap causes problems in other programs like chromium flatpak

**Comment by paju1986** [comment]
Thanks you that works! 
but that means i have to always run sunshine as sudo ?

**Comment by linuxkernel94** [comment]
I'm facing the same issue on Fedora Silverblue


```
Fatal: You must run [sudo setcap cap_sys_admin+p $(readlink -f $(which sunshine))] for KMS display capture to work!
Fatal: Unable to find display or encoder during startup.
Fatal: Please check that a display is connected and powered on.
```
